system/lib/pthread/emscripten_futex_wake.c

```diff
@@ -32,13 +30,9 @@ int emscripten_futex_wake(volatile void *addr, int count) {
   // this scheme does not adhere to real queue-based waiting.
   int main_thread_woken = 0;
   if (a_cas_p(&_emscripten_main_thread_futex, (void*)addr, 0) == addr) {
-    // We only use __emscripten_main_thread_futex on the main browser thread,
-    // where we cannot block while we wait. Therefore we should only see it set
-    // from other threads (that should always support waiting), and not on the
-    // main thread itself. In other words, the main thread must never try to
-    // wake itself up!
-    assert(_emscripten_thread_supports_atomics_wait());
-    --count;
+    // The main browser thread must never try to wake itself up!
+    assert(!emscripten_is_main_browser_thread());
+    if (count != INT_MAX) --count;
     main_thread_woken = 1;
     if (count <= 0) {
```

Comment:
Maybe put this block inside the new `if (count != INT_MAX)` block?    We should also probably be consistent about using curly braces single line if blocks.

lgtm either way.

Reply:
Good point, updated.